### DISCUSSION TITLE
Is there an equivalent of getReactNativePersistence?

### DISCUSSION THREAD

**Discussion by Nantris** [invertase/react-native-firebase#7476] [Q&A] [answered]
Is this not needed when using React Native Firebase? I assume so but would appreciate clarification.

**Comment by mikehardy** [comment, marked as answer]
What is "getReactNativePersistence"?

https://stackoverflow.com/help/how-to-ask

**Reply by Nantris** [reply]
Apologies; you're right.

To provide context, [`getReactNativePersistence` ](https://firebase.google.com/docs/reference/js/auth.md#getreactnativepersistence)

**Reply by mikehardy** [reply]
Very interesting! Hadn't considered that, it doesn't seem to really make sense in a react-native-firebase context since we are always using react-native persistence (that is, lots of native stuff - it is not async-storage but it is *definitely* not a lightweight in-memory thing like the optional way of initializing dependencies with initializeAuth from firebase-js-sdk

In the react-native-firebase context, this is definitely not needed

The only slight modification I might make on that statement is for a different use case - as a package with the goal of being a drop-in replacement for firebase-ios-sdk we should implement it so that it exists, but it should be documented as unnecessary and only present for API / compile compatibility in codebases

**Reply by Nantris** [reply]
I might not be considering all of the possible use cases, but for us it's not necessary at all and having it implemented but not necessary would merely add complexity I think. You'll likely have better insights into the possible use cases.

My thinking is that a documentation-only approach is best.

By not needing to rely on `getReactNativePersistence` I was able to remove a mobile-only JS file and some function calls, and that always feels nice.
